graph/src/blockchain/mod.rs

```diff
@@ -135,6 +139,19 @@ impl ChainStoreBlock {
 //     Rpc(C::Client),
 // }
 
+/// A chain-agnostic interface for instantiating chains of a specific kind. Not
+/// all chains may support this interface, but that's okay, as some may require
+/// special logic.
+pub trait BuildableBlockchain: Blockchain {
```

Comment:
This comment is totally subjective so I will leave it just for your consideration.

I think I would rather have a the Blockchain as a type parameter. 

Also have a suggestion for the name 

```
pub trait BlockchainBuilder<C: Blockchain>
```

This means we could have multiple implementations of Builders if needed or a single one that's generic over Blockchain as opposed to having only 1 per Blockchain implementation. 

Other than that I don't have a huge preference and may also be missing something here.